pkg/healthcheck/healthcheck_output.go

```diff
@@ -48,20 +52,200 @@ func (cr CheckResults) RunChecks(observer CheckObserver) bool {
+				if len(stderr.String()) > 0 {
+					err = errors.New(stderr.String())
+				} else {
+					err = fmt.Errorf("invalid extension check output from \"%s\" (JSON object expected):\n%s\n[%s]", command, stdout.String(), err)
+				}
+				results.Results = append(results.Results, CheckResult{
+					Category:    CategoryID(extensionCmd),
+					Description: fmt.Sprintf("Running: %s", command),
+					Err:         err,
+					HintURL:     DefaultHintBaseURL + "extensions",
+				})
+				success = false
+			} else {
+				results.Results = append(results.Results, extensionResults.Results...)
+			}
+		}
+
+		var containsErr bool
+		for _, res := range results.Results {
+			if res.Err != nil {
+				containsErr = true
+				break
+			}
+		}
+
+		if output != JSONOutput && i < len(extensions) {
+			// add a new line to space out each check output
+			fmt.Fprintln(wout)
+			if output == ShortOutput && !containsErr {
+				fmt.Fprintln(wout, string(results.Results[0].Category))
+				fmt.Fprintln(wout, strings.Repeat("-", len(string(results.Results[0].Category))))
+			}
+		}
+
+		extensionSuccess := RunChecks(wout, werr, results, output)
+		if !extensionSuccess {
+			success = false
+		}
+	}
+
+	return success
+}
+
 // RunChecks runs the checks that are part of hc
 func RunChecks(wout io.Writer, werr io.Writer, hc Runner, output string) bool {
 	if output == JSONOutput {
 		return runChecksJSON(wout, werr, hc)
 	}
+
+	if output == ShortOutput {
+		return runChecksShort(wout, hc)
+	}
+
 	return runChecksTable(wout, hc)
 }
 
-func runChecksTable(wout io.Writer, hc Runner) bool {
+func runChecksShort(wout io.Writer, hc Runner) bool {
 	var lastCategory CategoryID
+
 	spin := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
 	spin.Writer = wout
 
 	prettyPrintResults := func(result *CheckResult) {
+		spin.Stop()
+		if result.Retry {
+			if isatty.IsTerminal(os.Stdout.Fd()) {
+				spin.Suffix = fmt.Sprintf(" %s", result.Err)
+				spin.Color("bold") // this calls spin.Restart()
+			}
+			return
+		}
+
+		if result.Err == nil {
+			return
+		}
+
+		status := failStatus
+		if result.Warning {
+			status = warnStatus
+		}
+
+		if lastCategory != result.Category {
+			if lastCategory != "" {
+				fmt.Fprintln(wout)
+			}
+
+			fmt.Fprintln(wout, result.Category)
+			fmt.Fprintln(wout, strings.Repeat("-", len(result.Category)))
+
+			lastCategory = result.Category
+		}
+
+		fmt.Fprintf(wout, "%s %s\n", status, result.Description)
+		if result.Err != nil {
+			fmt.Fprintf(wout, "    %s\n", result.Err)
+			if result.HintURL != "" {
+				fmt.Fprintf(wout, "    see %s for hints\n", result.HintURL)
+			}
+		}
+	}
+
+	success := hc.RunChecks(prettyPrintResults)
+	// this empty line separates final results from the checks list in the output
+	fmt.Fprintln(wout, "")
+
+	if !success {
+		fmt.Fprintf(wout, "Status check results are %s\n", failStatus)
+	} else {
+		fmt.Fprintf(wout, "Status check results are %s\n", okStatus)
```

Comment:
Would it be worth it to include a warning count in the `--short` check? wdyt

i.e if we finished with a successful status but we logged some warnings, would it be worth it to say `3 warnings found` or something similar?

Reply:
Good idea. I'm on the fence with this one since we already print out the warnings if we encounter them. However, have a enumerated summary of the check results might be useful in getting an brief idea of how many things passed vs failed vs succeeded.